Mathlib/Topology/Bases.lean

```diff
@@ -119,6 +120,39 @@ theorem isTopologicalBasis_of_subbasis {s : Set (Set α)} (hs : t = generateFrom
     exact ⟨{t}, ⟨finite_singleton t, singleton_subset_iff.2 ht⟩, rfl⟩
 #align topological_space.is_topological_basis_of_subbasis TopologicalSpace.isTopologicalBasis_of_subbasis
 
+theorem isTopologicalBasis_of_subbasis_of_finiteInter {s : Set (Set α)} (hsg : t = generateFrom s)
+    (hsi : FiniteInter s) : IsTopologicalBasis s := by
+  have e1 {s : Set (Set α)} (hsi : FiniteInter s) :
+    (fun g => ⋂₀ g) '' { f : Set (Set α) | f.Finite ∧ f ⊆ s } = s := by
+    rw [le_antisymm_iff]
+    constructor
+    · intro F
+      simp only [mem_image, mem_setOf_eq, forall_exists_index, and_imp]
+      intro G hG hGs hi
+      rw [← hi]
+      let G' := Finite.toFinset hG
```

Comment:
I'm sure there must be a better argument, but I haven't been able to figure it out.